### ISSUE TITLE
Slow build -> deploy on iOS

### ISSUE THREAD

**Issue by Inrego** [dotnet/maui#14560]
### Description

The build process of iOS apps is very slow. Almost a minute (55 seconds) until first page is shown on physical device. Especially the build seems to take up much time, clocking in at around 40 seconds.

I'm deploying to a physical device that is connected by cable to the Mac.

Here are the specs of the Mac:

![image](https://user-images.githubusercontent.com/2242329/231741139-5071090a-085d-4f69-8a43-249f6a3e3732.png)

I tried the same on Xamarin.Forms, where the same process takes 45 seconds, so a bit faster but also tedious.

Such a long cycle to redeploy simple changes in my code makes it very tedious to develop on iOS.
And no, I'm not using Hot Reload, because SkiaSharp doesn't work on iOS with Hot Reload.

### Steps to Reproduce

1. Create new .NET MAUI app from template (.NET 7)
2. Set up connection to Mac
3. Set up automatic provisioning
4. Plug physical device into Mac
5. Deploy app to device
6. Stop app
7. Make small change in code
8. Deploy app again

### Link to public reproduction project repository

N/A

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iPadOS 13.6.1

**Comment by TomDacquinDMS** [comment]
55 seconds? You're lucky, here it takes 5 minutes (same setup as you Visual Studio connect to MacbookPro, which has a physical iPhone connected via cable)

**Comment by drasticactions** [comment]
Have you tried building and deploying a iOS app from Xcode with that setup to see how much faster it is?

**Comment by Inrego** [comment]
> Have you tried building and deploying a iOS app from Xcode with that setup to see how much faster it is?

I just tried that. It was around 6 seconds.

**Comment by Inrego** [comment]
Also, I want to specify a bit: I'm developing in Visual Studio on Windows with connection to a Mac Mini on my desk.
The timings I've done are not for first build, which I can understand would take longer. It's for "incremental" builds, where I'm just changing a single character in the UI and then hitting F5. All timings have been done on the project template which is created automatically when creating a new .NET MAUI app project.

I just tried a .NET MAUI app directly on the mac with Visual Studio for Mac, and it took just under 30 seconds.

So to recap:

Windows -> Mac: 55 seconds
Directly on Mac: 30 seconds
Swift Project in Xcode: 6 seconds

But yeah.. it's a pain to develop for iOS with .NET MAUI, when you have to wait a minute every time you changed something.

**Comment by w1ggor** [comment]
I have a Mac Mini taking at least 40 minutes on release builds, almost sleeping throughout the builds.

**Comment by MarkMcCormackSF** [comment]
You should try adding some additional libraries into your project too.  My project isn't that big and the development cycle time is about 5 minutes.  I don't see how it's practical to develop realistically with such a slow development cycle for every minor change to the app DLL.

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.9.0 Preview 2. Can repro on iOS platforms with sample project.

**Comment by jfversluis** [comment]
Hey everyone! It's been a minute. Are there still people on here that experience this? Has it gotten better, worse? Is there still anything we can do or investigate here?

cc: @rolfbjarne @dalexsoto

**Comment by rolfbjarne** [comment]
It would be useful to know how slow it is in comparison when building directly on the Mac (using VSCode) vs building Windows. When using Windows, the network conditions can heavily affect the build times, because a lot of data has to be copied over the network.

Then it's also useful to get [MSBuild binlogs](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) for slow builds, because binlogs contain timing which will show us exactly where the time is spent.

**Comment by Inrego** [comment]
@rolfbjarne @jfversluis  I don't believe this is fixed, although I'm quite busy at work at the moment, so I don't have time to do the timings again. I've already provided all relevant timings earlier. 

Since you think it might be fixed, I'm curious to know what's the fastest build time you are able to achieve for a full build. It's not some edge case I'm reporting. It should be easy to reproduce. Just create a new MAUI app from default template on Windows, and build for iOS using a Mac Agent.

**Comment by Inrego** [comment]
Regarding your comment about network speed. The mac is sitting on my desk, connected to my windows machine with a gigabit switch. A new MAUI project takes up 22 MB disk space. I do not believe latency or transfer speed to be the issue.

With that said, I do believe there are improvements to be made in the tooling that takes care of this. Because according to the timings I posted earlier, it is 25 seconds faster to build directly on the Mac, compared to over a network with Mac Agent. I believe there's gotta be room for optimizations in this area.

But I do sincerely hope there's also room for improvements/optimization for the build itself on the Mac. 30 seconds for each build makes iterative development a pain. As mentioned earlier, a Swift project builds in just 6 seconds on the same Mac.

And note that these timings are with just the template project with no added dependencies or code. So in real-world scenarios, the build times are even longer.

**Comment by pierre01** [comment]
I still have the same issue with the latest Version 17.12.0 Preview 2.1
Every time I compile, the execution on the IOS emulator takes longer and longer to up to 5 minutes.

**Comment by safwankarim-sentient** [comment]
[UPDATE]

I followed some recommendations on this issue and it worked amazing! I added both 
`-p:EnableAssemblyILStripping=false -p:MtouchUseLlvm=false` to my publish command.

[issues-25022](https://github.com/dotnet/maui/issues/25022#issuecomment-2385334527)

![Image](https://github.com/user-attachments/assets/97108552-09ad-4141-a9aa-b34d5ae386a4)
